### DISCUSSION TITLE
Misidentify table contents

### DISCUSSION THREAD

**Discussion by xer23452** [remarkjs/remark#1352] [Q&A]
<img width="961" alt="image" src="https://github.com/user-attachments/assets/6ac97944-0d9f-44d1-a6c4-079c099ddae4">
The last sentence is not part of the table, but is recognized as the last line 

error screenshot：
<img width="631" alt="image" src="https://github.com/user-attachments/assets/9e5e56f6-3c41-4ce9-873e-07c8babf1fdb">

**Comment by ChristianMurphy** [comment]
@13832198312 can you share a reproducible example of what you think is an issue?
Using https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-vite as a starting point?

**Reply by xer23452** [reply]
<img width="1131" alt="image" src="https://github.com/user-attachments/assets/e798efd7-5e25-4922-8896-1534eac9c3f8">
still has the problem

**Reply by xer23452** [reply]
the td will be rendered when the end of markdownSource is "|"，but now there are no "|" in the end

**Reply by wooorm** [reply]
There is no problem. This is how GFM works.

If you still think there are problems: **do not share screenshots**. Share actually runnable code. Make your code **small**. Make the tiniest possible case.

**Reply by xer23452** [reply]
```
import Markdown from 'react-markdown';
import remarkGfm from 'remark-gfm';
const markdownSource = `
以下是宝马和奔驰的对比表格：\n|       | A | B |\n|------------|-------------------------------|-------------------------------|\n| 1   | 123 | 789 |\n| 2       | melon |  apple| 
ss
`;

function App() {
  return <Markdown remarkPlugins={[remarkGfm]}>{markdownSource}</Markdown>;
}

export default App;

```

**Reply by wooorm** [reply]
Make it smaller.

Still, try it out. Reiterating: this is how GFM works:

以下是宝马和奔驰的对比表格：
|       | A | B |
|------------|-------------------------------|-------------------------------|
| 1   | 123 | 789 |
| 2       | melon |  apple| 
ss

**Reply by xer23452** [reply]
but i hope “ss” is not table content， If I put a “|” after “ss”, it will be the table content  and there is no "|" in the end
<img width="787" alt="image" src="https://github.com/user-attachments/assets/38cfa6d5-515e-425d-830d-9e682daeb3c2">

**Reply by wooorm** [reply]
And I want world peace.

This is how tables work.

This is how markdown works.

Use a blank line:

```markdown
| a | b |
| - | - |
c

| d | e |
| - | - |

f
```

| a | b |
| - | - |
c

| d | e |
| - | - |

f

**Reply by xer23452** [reply]
but i can not control the data source, and other markdown tools can identify right

**Reply by wooorm** [reply]
I cannot help you.
The data is wrong.
The other tool is wrong.
